### ISSUE TITLE
[TF 2.0] unconnected_gradients = 'zero' does not work

### ISSUE THREAD

**Issue by alper111** [tensorflow/tensorflow#26808]
**System information**
- OS Platform and Distribution: MacOS 10.14.3 
- TensorFlow installed from binary
- TensorFlow version: 2.0.0a0
- Python version: 3.7.2

I try to get gradients w.r.t. model parameters. Though I was getting None values. Here is an example:

```
> import tensorflow as tf
> import tensorflow.keras.layers as layers

> model = tf.keras.Sequential()
> model.add(layers.Dense(10, input_shape=(2,)))
> with tf.GradientTape() as tape:
>   loss = tf.random.normal((10, 10))
> grads = tape.gradient(loss, model.trainable_variables, unconnected_gradients='zero')
> print(grads)
[None, None]
```

I expect these values to be zero. Though they are not.

**Comment by saxenasaurabh** [comment]
We need to add special treatment for DT_RESOURCE tensors when building the zeros tensors [here](https://github.com/tensorflow/tensorflow/blob/919b38007ea755a5b5ec87af324c91f55dce6717/tensorflow/python/eager/pywrap_tfe_src.cc#L1764). Maybe we can return a float32 with zeros here? @akshaym could you look into this?
